src/index.ts

```diff
@@ -10,25 +10,25 @@ import {pathToIndices} from "./utils";
  * @param entropy optional additional entropy
  */
 export function generateRandomSecretKey(entropy?: Buffer): Buffer {
-    let ikm = randomBytes(16);
-    if(entropy) {
-        ikm = Buffer.concat([entropy, ikm]);
-    }
-    return deriveMasterSK(ikm);
+  let ikm = randomBytes(16);
+  if(entropy) {
+    ikm = Buffer.concat([entropy, ikm]);
+  }
+  return deriveMasterSK(ikm);
 }
 
 export function mnemonicToSecretKey(mnemonic: string, path: string | null = "m/12381/60/0/0"): Buffer {
```

Comment:
I would love to see an exported function like `deriveKey(key: Buffer, path: string): Buffer` (or named similarly) that could also be used in this function.
eg:
`return path ? deriveKey(masterKey, path) : masterKey;`